File: test/files/burrow.vb
' VB.NET with Elan 2.0.0-beta4

Sub main()
  Dim blocks = New BlockGraphics() ' variable definition
  Dim x = 20 ' variable definition
  Dim y = 15 ' variable definition
  While True
    blocks.put(x, y, red) ' procedure call
    displayBlockGraphics(blocks) ' procedure call
    blocks.put(x, y, black) ' procedure call
    Dim direction = randint(0, 3) ' variable definition
    If direction = 0 Then
      x = min({x + 1, 39}) ' assignment
    ElseIf direction = 1 Then
      x = max({x - 1, 0}) ' assignment
    ElseIf direction = 2 Then
      y = min({y + 1, 29}) ' assignment
    ElseIf direction = 3 Then
      y = max({y - 1, 0}) ' assignment
    End If
  End While
End Sub

File: test/files/burrow.java
// Java with Elan 2.0.0-beta4

public class Global {

static void main() {
  var blocks = new BlockGraphics();
  var x = 20;
  var y = 15;
  while (true) {
    blocks.put(x, y, red); // procedure call
    blocks.display(); // procedure call
    blocks.put(x, y, black); // procedure call
    var direction = randint(0, 3);
    if (direction == 0) {
      x = min(list(x + 1, 39)); // assignment
    } else if (direction == 1) {
      x = max(list(x - 1, 0)); // assignment
    } else if (direction == 2) {
      y = min(list(y + 1, 29)); // assignment
    } else if (direction == 3) {
      y = max(list(y - 1, 0)); // assignment
    } // end if
  } // end while
} // end main
} // end Global
